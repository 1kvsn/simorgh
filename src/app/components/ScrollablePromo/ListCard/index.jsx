import React, { useContext } from 'react';
import styled from '@emotion/styled';
import { GEL_SPACING_TRPL } from '@bbc/gel-foundations/spacings';
import { array } from 'prop-types';
import SingleCard from '../SingleCard';
import { ServiceContext } from '#contexts/ServiceContext';

const ScrollPromo = styled.ul`
  padding-bottom: ${GEL_SPACING_TRPL};
  display: flex;
  overflow-wrap: break-word;
  gap: 16px;

  list-style: none;
  ${({ dir }) => `padding-${dir === 'ltr' ? 'left' : 'right'}: 0;`}
  margin: 0;

  overflow-x: scroll;

  /* Avoid using smooth scrolling as it causes accessibility issues */
  scroll-behavior: auto;
  -webkit-overflow-scrolling: touch;

  /* Hide scrollbar */
  scrollbar-width: none;
  -ms-overflow-style: none;
  &::-webkit-scrollbar {
    display: none;
  }
`;

const StyledList = styled.li`
  flex-shrink: 0;
  width: 205px;
  background-color: #ffffff;
  padding: 16px;
`;

const ListCard = ({ blocks }) => {
  const { dir } = useContext(ServiceContext);
  const threeblocks = blocks.slice(1, 4);

  return (
    <ScrollPromo dir={dir} role="list">
      {threeblocks.map((block, index) => {
        return (
          <StyledList>
            <SingleCard
              block={block}
              // eslint-disable-next-line react/no-array-index-key
              key={index}
            />
          </StyledList>
        );
      })}
    </ScrollPromo>
  );
};

ListCard.propTypes = {
  blocks: array.isRequired,
};

export default ListCard;
