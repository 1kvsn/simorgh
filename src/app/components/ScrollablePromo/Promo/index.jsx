import React, { useContext } from 'react';
import { object } from 'prop-types';
import styled from '@emotion/styled';
import pathOr from 'ramda/src/pathOr';
import { getPica } from '@bbc/gel-foundations/dist/typography';
import { getSerifRegular } from '@bbc/psammead-styles/dist/font-styles';
import { C_GREY_6 } from '@bbc/psammead-styles/dist/colours';
import {
  GEL_SPACING_DBL,
  GEL_SPACING_TRPL,
} from '@bbc/gel-foundations/dist/spacings';
import {
  GEL_GROUP_0_SCREEN_WIDTH_MIN,
  GEL_GROUP_3_SCREEN_WIDTH_MIN,
  GEL_GROUP_4_SCREEN_WIDTH_MIN,
} from '@bbc/gel-foundations/dist/breakpoints';
import { ServiceContext } from '#contexts/ServiceContext';
import filterForBlockType from '#lib/utilities/blockHandlers';

const C_GREY_8 = '#202224';

const Link = styled.a`
  ${({ script }) => script && getPica(script)}
  ${({ service }) => service && getSerifRegular(service)}
  width: 100%;

  text-overflow: ellipsis;
  text-decoration: none;

  overflow: hidden;
  display: -webkit-box;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;

  &:hover,
  &:focus {
    text-decoration: underline;
  }

  color: ${C_GREY_8};
  &:visited {
    color: ${C_GREY_6};
  }
`;

const PromoBox = styled.div`
  margin-bottom: ${GEL_SPACING_TRPL};
  background-color: #ffffff;
  padding: ${GEL_SPACING_DBL};
  @media (min-width: ${GEL_GROUP_0_SCREEN_WIDTH_MIN}) {
    width: 237px;
  }
  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
    width: 178px;
  }
  @media (min-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN}) {
    width: 205px;
  }
`;

const Promo = ({ block }) => {
  const { script, service, dir } = useContext(ServiceContext);
  const textBlock = filterForBlockType(
    pathOr({}, ['model', 'blocks'], block),
    'text',
  );
  const href = pathOr(
    '',
    ['model', 'blocks', '0', 'model', 'blocks', '0', 'model', 'locator'],
    textBlock,
  );
  const title = pathOr(
    '',
    ['model', 'blocks', '0', 'model', 'blocks', '0', 'model', 'text'],
    textBlock,
  );
  return (
    <PromoBox dir={dir}>
      <Link href={href} service={service} script={script} dir={dir}>
        {title}
      </Link>
    </PromoBox>
  );
};

Promo.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  block: object.isRequired,
};

export default Promo;
