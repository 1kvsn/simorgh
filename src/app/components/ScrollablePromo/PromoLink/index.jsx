import React, { useContext } from 'react';
import { object } from 'prop-types';
import styled from '@emotion/styled';
import { pathOr } from 'ramda';

import { getPica } from '@bbc/gel-foundations/dist/typography';
import { getSerifRegular } from '@bbc/psammead-styles/dist/font-styles';
import { C_GREY_6 } from '@bbc/psammead-styles/dist/colours';

import { ServiceContext } from '#contexts/ServiceContext';
import filterForBlockType from '#lib/utilities/blockHandlers';

const C_GREY_8 = '#202224';

//Use Reith Medium instead of Reith Bold
const Link = styled.a`
  ${({ script }) => script && getPica(script)};
  ${({ service }) => service && getSerifRegular(service)};
  width: 100%;

  text-overflow: ellipsis;
  text-decoration: none;

  overflow: hidden;
  display: -webkit-box;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;

  &:hover,
  &:focus {
    text-decoration: underline;
  }

  color: ${C_GREY_8};
  &:visited {
    color: ${C_GREY_6};
  }
`;

const PromoLink = ({ block }) => {
  const { script, service, dir } = useContext(ServiceContext);
  console.log(getSerifRegular(service));
  const textBlock = filterForBlockType(
    pathOr({}, ['model', 'blocks'], block),
    'text',
  );
  const href = pathOr(
    '',
    ['model', 'blocks', '0', 'model', 'blocks', '0', 'model', 'locator'],
    textBlock,
  );
  const title = pathOr(
    '',
    ['model', 'blocks', '0', 'model', 'blocks', '0', 'model', 'text'],
    textBlock,
  );
  return (
    <Link href={href} service={service} script={script} dir={dir}>
      {title}
    </Link>
  );
};

PromoLink.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  block: object.isRequired,
};

export default PromoLink;
