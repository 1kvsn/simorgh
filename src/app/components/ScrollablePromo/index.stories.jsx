/* eslint-disable react/prop-types */
import React from 'react';
import { withServicesKnob } from '@bbc/psammead-storybook-helpers';
import { withKnobs } from '@storybook/addon-knobs';
import styled from '@emotion/styled';
import mundoRecommendationsData from '#pages/StoryPage/fixtureData/recommendations.ltr.json';
import { ServiceContextProvider } from '#contexts/ServiceContext';
import { ToggleContextProvider } from '#contexts/ToggleContext';
import ScrollablePromo from '.';
import {
  threeLinks,
  oneLinkOnly,
  moreThanThreeLinks,
  twoLinksWithNoImages,
  truncatedTextInSingleLink,
  arabicText,
} from './helpers/fixtureData';

const BackGround = styled.div`
  background-color: #f6f6f6;
  padding: 2rem;
`;

const ScrollablePromoComponent = ({
  data,
  service,
  script,
  dir,
  translations,
  recommendations,
}) => (
  <ToggleContextProvider>
    <BackGround>
      <ServiceContextProvider
        service={service}
        script={script}
        dir={dir}
        translations={translations}
      >
        <ScrollablePromo blocks={data} recommendations={recommendations} />
      </ServiceContextProvider>
    </BackGround>
  </ToggleContextProvider>
);

export default {
  title: 'components/scrollable promo component',
  ScrollablePromoComponent,
  decorators: [withKnobs, withServicesKnob()],
};

export const ThreeLinks = props => (
  <ScrollablePromoComponent data={threeLinks} {...props} />
);

export const OnlyOneLink = props => (
  <ScrollablePromoComponent data={oneLinkOnly} {...props} />
);

export const MoreThanThreeLinks = props => (
  <ScrollablePromoComponent data={moreThanThreeLinks} {...props} />
);

export const NoImagesInData = props => (
  <ScrollablePromoComponent data={twoLinksWithNoImages} {...props} />
);

export const TruncatedTextInSingleLink = props => (
  <ScrollablePromoComponent data={truncatedTextInSingleLink} {...props} />
);

export const ArabicText = props => (
  <ScrollablePromoComponent data={arabicText} {...props} />
);

export const Recommendations = props => (
  <ScrollablePromoComponent
    data={mundoRecommendationsData}
    recommendations
    {...props}
  />
);
