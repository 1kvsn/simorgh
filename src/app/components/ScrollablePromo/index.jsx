import React, { useContext } from 'react';
import { array } from 'prop-types';

import {
  GEL_SPACING_DBL,
  GEL_SPACING,
} from '@bbc/gel-foundations/dist/spacings';

import {
  GEL_GROUP_2_SCREEN_WIDTH_MIN,
  GEL_GROUP_3_SCREEN_WIDTH_MIN,
  GEL_GROUP_4_SCREEN_WIDTH_MIN,
} from '@bbc/gel-foundations/dist/breakpoints';

import Promo from './Promo';
import PromoList from './PromoList';

import { GridItemMediumNoMargin } from '#app/components/Grid';

import { ServiceContext } from '#contexts/ServiceContext';

const ScrollablePromo = ({ blocks }) => {
  const { dir } = useContext(ServiceContext);
  const isSingleItem = blocks.length === 2;

  const promoBox = `
    flex-shrink: 0;
    width: 237px;
    background-color: #ffffff;
    padding: ${GEL_SPACING_DBL};
    margin: 0;
    margin-${dir === 'ltr' ? 'left' : 'right'}: ${GEL_SPACING};
    margin-${dir === 'ltr' ? 'right' : 'left'}: ${GEL_SPACING};

    @media (min-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}){
      &:first-child {
        margin-${dir === 'ltr' ? 'left' : 'right'}: ${GEL_SPACING_DBL};
      }
    }

    @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}){
      width: 178px;
    }

    @media (min-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN}){
      &:first-child {
        margin:0;
      }
      width: 211px;
    }
  `;

  // IF NO PROMO RETURN NULL
  return (
    <GridItemMediumNoMargin>
      {isSingleItem ? (
        <Promo block={blocks[1]} Ourstyle={promoBox} />
      ) : (
        <PromoList blocks={blocks} Ourstyle={promoBox} />
      )}
    </GridItemMediumNoMargin>
  );
};

ScrollablePromo.propTypes = {
  blocks: array.isRequired,
};

export default ScrollablePromo;
