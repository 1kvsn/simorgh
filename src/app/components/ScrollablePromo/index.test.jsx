/* eslint-disable react/prop-types */
import React from 'react';
import { render } from '@testing-library/react';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import * as viewTracking from '#hooks/useViewTracker';
import { ToggleContextProvider } from '#contexts/ToggleContext';
import { ServiceContext } from '#contexts/ServiceContext';
import * as clickTracking from '#hooks/useClickTrackerHandler';
import mundoRecommendationsData from '#pages/StoryPage/fixtureData/recommendations.ltr.json';
import {
  threeLinks,
  oneLinkOnly,
  moreThanThreeLinks,
} from './helpers/fixtureData';
import ScrollablePromo from '.';
import { edOjA, edOjB } from './fixtures';

const ScrollablePromoWithContext = ({
  blocks,
  blockGroupIndex,
  recommendations,
  translations,
  service,
}) => (
  <ToggleContextProvider>
    <ServiceContext.Provider
      value={{
        service,
        translations,
      }}
    >
      <ScrollablePromo
        blocks={blocks}
        blockGroupIndex={blockGroupIndex}
        recommendations={recommendations}
      />
    </ServiceContext.Provider>
  </ToggleContextProvider>
);

describe('ScrollablePromo', () => {
  it('should return null if no data is passed', () => {
    const { container } = render(
      <ScrollablePromoWithContext blocks={{}} service="news" />,
    );
    expect(container).toBeEmptyDOMElement();
  });

  it('should render max 3 promo items', () => {
    const { getAllByRole } = render(
      <ScrollablePromoWithContext blocks={moreThanThreeLinks} service="news" />,
    );
    expect(getAllByRole('listitem').length).toEqual(3);
  });

  it('should render single promo item', () => {
    const { container } = render(
      <ScrollablePromoWithContext blocks={oneLinkOnly} service="news" />,
    );
    expect(container.childElementCount).toEqual(1);
  });

  it('should not render a list when there is only one promo', () => {
    const { queryByRole } = render(
      <ScrollablePromoWithContext blocks={oneLinkOnly} service="news" />,
    );

    expect(queryByRole('list')).not.toBeInTheDocument();
    expect(queryByRole('listitem')).not.toBeInTheDocument();
  });

  it('should render unordered list if more than 1 item', () => {
    const { queryByRole, getAllByRole } = render(
      <ScrollablePromoWithContext blocks={threeLinks} service="news" />,
    );
    expect(queryByRole('list')).toBeInTheDocument();
    expect(getAllByRole('listitem').length).toEqual(3);
  });

  describe('event tracking in editorial onward journeys', () => {
    afterEach(() => {
      jest.clearAllMocks();
    });

    it('should call the view tracking hook with the correct params with one editorial onward journey', () => {
      const viewTrackerSpy = jest.spyOn(viewTracking, 'default');

      render(
        <ScrollablePromoWithContext
          blocks={edOjA.model.blocks}
          blockGroupIndex={1}
          service="news"
        />,
      );

      expect(viewTrackerSpy).toHaveBeenCalledWith({
        componentName: 'edoj1',
        format: 'CHD=edoj',
      });
    });

    it('should call the view tracking hook with the correct params with multiple editorial onward journeys', () => {
      const viewTrackerSpy = jest.spyOn(viewTracking, 'default');
      render(
        <ScrollablePromoWithContext
          blocks={edOjA.model.blocks}
          blockGroupIndex={1}
          service="news"
        />,
      );
      render(
        <ScrollablePromoWithContext
          blocks={edOjB.model.blocks}
          blockGroupIndex={2}
          service="news"
        />,
      );

      expect(viewTrackerSpy).toHaveBeenCalledTimes(2);
      expect(viewTrackerSpy).toHaveBeenCalledWith({
        componentName: 'edoj1',
        format: 'CHD=edoj',
      });
      expect(viewTrackerSpy).toHaveBeenCalledWith({
        componentName: 'edoj2',
        format: 'CHD=edoj',
      });
    });

    it('should call the click tracking hook with one editorial onward journey', () => {
      const clickTrackerSpy = jest.spyOn(clickTracking, 'default');
      render(
        <ScrollablePromoWithContext
          blocks={edOjA.model.blocks}
          blockGroupIndex={1}
          service="news"
        />,
      );

      expect(clickTrackerSpy).toHaveBeenCalledWith({
        componentName: 'edoj1',
        format: 'CHD=edoj',
      });
    });

    it('should call the click tracking hook with multiple editorial onward journeys', () => {
      const clickTrackerSpy = jest.spyOn(clickTracking, 'default');
      render(
        <ScrollablePromoWithContext
          blocks={edOjA.model.blocks}
          blockGroupIndex={1}
          service="news"
        />,
      );
      render(
        <ScrollablePromoWithContext
          blocks={edOjB.model.blocks}
          blockGroupIndex={2}
          service="news"
        />,
      );

      expect(clickTrackerSpy).toHaveBeenCalledTimes(2);
      expect(clickTrackerSpy).toHaveBeenCalledWith({
        componentName: 'edoj1',
        format: 'CHD=edoj',
      });
      expect(clickTrackerSpy).toHaveBeenCalledWith({
        componentName: 'edoj2',
        format: 'CHD=edoj',
      });
    });
  });

  shouldMatchSnapshot(
    'it should match a11y snapshot for single card',
    <ScrollablePromoWithContext blocks={oneLinkOnly} service="news" />,
  );

  shouldMatchSnapshot(
    'it should match a11y snapshot for list',
    <ScrollablePromoWithContext blocks={threeLinks} service="news" />,
  );
});

describe('recommendationEOJ', () => {
  it('should render recommendation variation when recommendation && recommendation data are passed', () => {
    const { getAllByRole, queryByRole, queryByText } = render(
      <ScrollablePromoWithContext
        blocks={mundoRecommendationsData}
        recommendations
        service="news"
      />,
    );
    expect(getAllByRole('listitem').length).toEqual(3);
    expect(queryByRole('region')).toBeInTheDocument();
    expect(queryByText('Related stories')).toBeInTheDocument();
  });

  it('should render translated title', () => {
    const { queryByText } = render(
      <ScrollablePromoWithContext
        blocks={mundoRecommendationsData}
        recommendations
        service="mundo"
        translations={{ relatedContent: 'contenido relacionado' }}
      />,
    );
    expect(queryByText('contenido relacionado')).toBeInTheDocument();
  });

  shouldMatchSnapshot(
    'it should match a11y snapshot for recommendationEOJ',
    <ScrollablePromoWithContext
      blocks={mundoRecommendationsData}
      recommendations
      service="news"
    />,
  );
});
