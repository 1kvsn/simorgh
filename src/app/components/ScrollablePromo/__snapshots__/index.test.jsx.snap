// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollablePromo it should match a11y snapshot for list 1`] = `
@media (max-width: 14.9375rem) {
  .emotion-0 {
    margin-left: 0%;
  }
}

@media (min-width: 15rem) and (max-width: 24.9375rem) {
  .emotion-0 {
    margin-left: 0%;
  }
}

@media (min-width: 25rem) and (max-width: 37.4375rem) {
  .emotion-0 {
    margin-left: 0%;
  }
}

@media (min-width: 37.5rem) and (max-width: 62.9375rem) {
  .emotion-0 {
    margin-left: 0%;
  }
}

@media (min-width: 63rem) and (max-width: 79.9375rem) {
  .emotion-0 {
    margin-left: 20%;
  }
}

@media (min-width: 80rem) {
  .emotion-0 {
    margin-left: 40%;
  }
}

@supports (display: grid) {
  .emotion-0 {
    display: block;
    width: initial;
    margin: 0;
  }

  @media (max-width: 14.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 15rem) and (max-width: 24.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 25rem) and (max-width: 37.4375rem) {
    .emotion-0 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 37.5rem) and (max-width: 62.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(5, 1fr);
      grid-column-end: span 5;
      grid-column-start: 1;
    }
  }

  @media (min-width: 63rem) and (max-width: 79.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(5, 1fr);
      grid-column-end: span 5;
      grid-column-start: 2;
    }
  }

  @media (min-width: 80rem) {
    .emotion-0 {
      grid-template-columns: repeat(10, 1fr);
      grid-column-end: span 10;
      grid-column-start: 5;
    }
  }
}

.emotion-2 {
  list-style: none;
  padding-right: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow-x: scroll;
  scroll-behavior: auto;
  -webkit-overflow-scrolling: touch;
  scrollbar-width: none;
  -ms-overflow-style: none;
}

.emotion-2::-webkit-scrollbar {
  display: none;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

@media (min-width: 0rem) {
  .emotion-4 {
    margin-right: 0.5rem;
  }

  .emotion-4:first-child {
    margin-right: 0.5rem;
  }

  .emotion-4:last-child {
    margin-left: 0.5rem;
  }
}

@media (min-width: 25rem) {
  .emotion-4 {
    margin-right: 1rem;
  }

  .emotion-4:first-child {
    margin-right: 1rem;
  }
}

@media (min-width: 63rem) {
  .emotion-4 {
    margin-right: 1rem;
  }

  .emotion-4:first-child {
    margin-right: 0;
  }
}

.emotion-6 {
  background-color: #FFFFFF;
  padding: 1rem;
  margin-bottom: 1.5rem;
}

@media (min-width: 0rem) {
  .emotion-6 {
    width: 14.8125rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-6 {
    width: 11.125rem;
  }
}

@media (min-width: 63rem) {
  .emotion-6 {
    width: 12.6875rem;
  }
}

.emotion-8 {
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 700;
  font-style: normal;
  width: 100%;
  overflow-wrap: break-word;
  text-overflow: ellipsis;
  -webkit-text-decoration: none;
  text-decoration: none;
  overflow-x: hidden;
  overflow-y: hidden;
  display: -webkit-box;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
  color: #202224;
}

.emotion-8:hover,
.emotion-8:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-8:visited {
  color: #545658;
}

<div>
  <div
    class="emotion-0 emotion-1"
    dir="ltr"
  >
    <ul
      class="emotion-2 emotion-3"
      role="list"
    >
      <li
        class="emotion-4 emotion-5"
      >
        <div
          class="emotion-6 emotion-7"
        >
          <a
            class="emotion-8 emotion-9"
            href="https://www.bbc.co.uk/pidgin"
          >
            Show headline
          </a>
        </div>
      </li>
      <li
        class="emotion-4 emotion-5"
      >
        <div
          class="emotion-6 emotion-7"
        >
          <a
            class="emotion-8 emotion-9"
            href="https://www.bbc.com/mundo"
          >
            Link with Headline
          </a>
        </div>
      </li>
      <li
        class="emotion-4 emotion-5"
      >
        <div
          class="emotion-6 emotion-7"
        >
          <a
            class="emotion-8 emotion-9"
            href="https://www.bbc.com/brasil"
          >
            Brasil homepage
          </a>
        </div>
      </li>
    </ul>
  </div>
</div>
`;

exports[`ScrollablePromo it should match a11y snapshot for single card 1`] = `
@media (max-width: 14.9375rem) {
  .emotion-0 {
    margin-left: 0%;
  }
}

@media (min-width: 15rem) and (max-width: 24.9375rem) {
  .emotion-0 {
    margin-left: 0%;
  }
}

@media (min-width: 25rem) and (max-width: 37.4375rem) {
  .emotion-0 {
    margin-left: 0%;
  }
}

@media (min-width: 37.5rem) and (max-width: 62.9375rem) {
  .emotion-0 {
    margin-left: 0%;
  }
}

@media (min-width: 63rem) and (max-width: 79.9375rem) {
  .emotion-0 {
    margin-left: 20%;
  }
}

@media (min-width: 80rem) {
  .emotion-0 {
    margin-left: 40%;
  }
}

@supports (display: grid) {
  .emotion-0 {
    display: block;
    width: initial;
    margin: 0;
  }

  @media (max-width: 14.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 15rem) and (max-width: 24.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 25rem) and (max-width: 37.4375rem) {
    .emotion-0 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 37.5rem) and (max-width: 62.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(5, 1fr);
      grid-column-end: span 5;
      grid-column-start: 1;
    }
  }

  @media (min-width: 63rem) and (max-width: 79.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(5, 1fr);
      grid-column-end: span 5;
      grid-column-start: 2;
    }
  }

  @media (min-width: 80rem) {
    .emotion-0 {
      grid-template-columns: repeat(10, 1fr);
      grid-column-end: span 10;
      grid-column-start: 5;
    }
  }
}

.emotion-2 {
  margin-right: 0.5rem;
}

@media (min-width: 25rem) {
  .emotion-2 {
    margin-right: 1rem;
  }
}

@media (min-width: 63rem) {
  .emotion-2 {
    margin-right: 0;
  }
}

.emotion-4 {
  background-color: #FFFFFF;
  padding: 1rem;
  margin-bottom: 1.5rem;
}

@media (min-width: 0rem) {
  .emotion-4 {
    width: 14.8125rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-4 {
    width: 11.125rem;
  }
}

@media (min-width: 63rem) {
  .emotion-4 {
    width: 12.6875rem;
  }
}

.emotion-6 {
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 700;
  font-style: normal;
  width: 100%;
  overflow-wrap: break-word;
  text-overflow: ellipsis;
  -webkit-text-decoration: none;
  text-decoration: none;
  overflow-x: hidden;
  overflow-y: hidden;
  display: -webkit-box;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
  color: #202224;
}

.emotion-6:hover,
.emotion-6:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-6:visited {
  color: #545658;
}

<div>
  <div
    class="emotion-0 emotion-1"
    dir="ltr"
  >
    <div
      class="emotion-2 emotion-3"
    >
      <div
        class="emotion-4 emotion-5"
      >
        <a
          class="emotion-6 emotion-7"
          href="https://www.bbc.com/pidgin"
        >
          Single link
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`recommendationEOJ it should match a11y snapshot for recommendationEOJ 1`] = `
@media (max-width: 14.9375rem) {
  .emotion-0 {
    margin-left: 0%;
  }
}

@media (min-width: 15rem) and (max-width: 24.9375rem) {
  .emotion-0 {
    margin-left: 0%;
  }
}

@media (min-width: 25rem) and (max-width: 37.4375rem) {
  .emotion-0 {
    margin-left: 0%;
  }
}

@media (min-width: 37.5rem) and (max-width: 62.9375rem) {
  .emotion-0 {
    margin-left: 0%;
  }
}

@media (min-width: 63rem) and (max-width: 79.9375rem) {
  .emotion-0 {
    margin-left: 20%;
  }
}

@media (min-width: 80rem) {
  .emotion-0 {
    margin-left: 40%;
  }
}

@supports (display: grid) {
  .emotion-0 {
    display: block;
    width: initial;
    margin: 0;
  }

  @media (max-width: 14.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 15rem) and (max-width: 24.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 25rem) and (max-width: 37.4375rem) {
    .emotion-0 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 37.5rem) and (max-width: 62.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(5, 1fr);
      grid-column-end: span 5;
      grid-column-start: 1;
    }
  }

  @media (min-width: 63rem) and (max-width: 79.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(5, 1fr);
      grid-column-end: span 5;
      grid-column-start: 2;
    }
  }

  @media (min-width: 80rem) {
    .emotion-0 {
      grid-template-columns: repeat(10, 1fr);
      grid-column-end: span 10;
      grid-column-start: 5;
    }
  }
}

.emotion-2 {
  display: block;
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 700;
  font-style: normal;
  font-size: 20px;
  margin-bottom: 1rem;
}

.emotion-4 {
  list-style: none;
  padding-right: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow-x: scroll;
  scroll-behavior: auto;
  -webkit-overflow-scrolling: touch;
  scrollbar-width: none;
  -ms-overflow-style: none;
}

.emotion-4::-webkit-scrollbar {
  display: none;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

@media (min-width: 0rem) {
  .emotion-6 {
    margin-right: 0.5rem;
  }

  .emotion-6:first-child {
    margin-right: 0;
  }

  .emotion-6:last-child {
    margin-left: 0.5rem;
  }
}

@media (min-width: 25rem) {
  .emotion-6 {
    margin-right: 1rem;
  }

  .emotion-6:first-child {
    margin-right: 0;
  }
}

@media (min-width: 63rem) {
  .emotion-6 {
    margin-right: 1rem;
  }

  .emotion-6:first-child {
    margin-right: 0;
  }
}

.emotion-8 {
  background-color: #FFFFFF;
  padding: 1rem;
  margin-bottom: 1.5rem;
}

@media (min-width: 0rem) {
  .emotion-8 {
    width: 14.8125rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-8 {
    width: 11.125rem;
  }
}

@media (min-width: 63rem) {
  .emotion-8 {
    width: 12.6875rem;
  }
}

.emotion-10 {
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 700;
  font-style: normal;
  width: 100%;
  overflow-wrap: break-word;
  text-overflow: ellipsis;
  -webkit-text-decoration: none;
  text-decoration: none;
  overflow-x: hidden;
  overflow-y: hidden;
  display: -webkit-box;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
  color: #202224;
}

.emotion-10:hover,
.emotion-10:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-10:visited {
  color: #545658;
}

<div>
  <section
    aria-labelledby="recommendations-heading"
    class="emotion-0 emotion-1"
    data-e2e="[object Object]"
    dir="ltr"
    role="region"
  >
    <strong
      class="emotion-2 emotion-3"
      id="recommendations-heading"
    >
      Related stories
    </strong>
    <ul
      class="emotion-4 emotion-5"
      role="list"
    >
      <li
        class="emotion-6 emotion-7"
      >
        <div
          class="emotion-8 emotion-9"
        >
          <a
            class="emotion-10 emotion-11"
            href="/mundo/noticias-53377054"
          >
            La conmovedora historia de cómo una madre y el hombre preso por la muerte de su hija se unieron para atrapar al verdadero asesino
          </a>
        </div>
      </li>
      <li
        class="emotion-6 emotion-7"
      >
        <div
          class="emotion-8 emotion-9"
        >
          <a
            class="emotion-10 emotion-11"
            href="/mundo/noticias-america-latina-52884902"
          >
            Coronavirus en Colombia | El caso de la vigilante Edy Fonseca: "Me retuvieron en un edificio de clase alta durante la pandemia, lo acepté y me siento culpable"
          </a>
        </div>
      </li>
      <li
        class="emotion-6 emotion-7"
      >
        <div
          class="emotion-8 emotion-9"
        >
          <a
            class="emotion-10 emotion-11"
            href="/mundo/noticias-internacional-53113381"
          >
            Soy una mujer genocida y aún me persiguen los recuerdos de lo que hice
          </a>
        </div>
      </li>
    </ul>
  </section>
</div>
`;
